code/modules/mob/living/silicon/robot/death.dm

```diff
@@ -66,3 +66,17 @@
 	sql_report_cyborg_death(src)
 
 	return ..(gibbed)
+
+//If there's an MMI in the robot, have it ejected when the mob goes away. --NEO
+/mob/living/silicon/robot/Destroy()
+	if(mmi && mind)//Safety for when a cyborg gets dust()ed. Or there is no MMI inside.
```

Comment:
Probably want to leave the `mind` check where it was, so that if somehow a borg has an MMI but no `mind` it won't fail to GC